freighter/py/freighter/websocket.py

```diff
@@ -117,17 +119,86 @@ async def __close_server(self, server_err: Exception | None):
         await self.__internal.close()
 
 
-DEFAULT_MAX_SIZE = 2**20
+DEFAULT_MAX_SIZE = 2 ** 20
 
 
-class WebsocketClient(AsyncMiddlewareCollector):
-    """An implementation of AsyncStreamClient that is backed by a websocket"""
-
-    __endpoint: URL
+class SyncWebsocketStream(Stream[RQ, RS]):
     __encoder: EncoderDecoder
-    __max_message_size: int
-    __secure: bool = False
-    __kwargs: dict[str, Any]
+    __internal: SyncClientProtocol
+    __server_closed: Exception | None
+    __send_closed: bool
+    __res_msg_t: Type[_Message[RS]]
+
+    def __init__(
+        self,
+        encoder: EncoderDecoder,
+        ws: SyncClientProtocol,
+        res_t: Type[RS],
+    ):
+        self.__encoder = encoder
+        self.__internal = ws
+        self.__send_closed = False
+        self.__server_closed = None
+        self.__res_msg_t = _new_res_msg_t(res_t)
+
+    def receive(self) -> tuple[RS | None, Exception | None]:
+        if self.__server_closed is not None:
+            return None, self.__server_closed
+
+        data = self.__internal.recv()
+        assert isinstance(data, bytes)
+        msg = self.__encoder.decode(data, self.__res_msg_t)
+
+        if msg.type == "close":
+            assert msg.error is not None
+            self.__close_server(decode_exception(msg.error))
+            return None, self.__server_closed
```

Comment:
Why should receiving a 'close' message on an open server return an exception after closing the server?

Reply:
Ah - I see, a `close` message is only sent when there's an exception.